mne/preprocessing/bads.py

```diff
@@ -47,3 +57,75 @@ def _find_outliers(X, threshold=3.0, max_iter=2, tail=0):
 
     bad_idx = np.where(my_mask)[0]
     return bad_idx
+
+
+def unify_bad_channels(insts):
+    """Unify bad channels across list of instances.
+
+    This function looks across the list of instances to gather a list of
+    "bad" channels. Every instance's info["bads"] is set with the same "bad"
+    channels.
+
+    Parameters
+    ----------
+    insts : list
+        List of instances (:class:`~mne.io.Raw`, :class:`~mne.Epochs`,
+        :class:`~mne.Evoked`, :class:`~mne.time_frequency.Spectrum`,
+        :class:`~mne.time_frequency.EpochSpectrum`) to unify bad channels.
+
+    Returns
+    -------
+    new_insts : list
+        List of instances with bad channels unified across instances.
+
+    Notes
+    -----
+    This function operates in-place.
+
+    .. versionadded:: 1.6
+    """
+    # check input is not an empty list
+    if len(insts) == 0:
+        raise ValueError("Be sure insts is not empty list")
+
+    # first check that each object is mne object
+    inst_type = type(insts[0])
+    valid_types = (BaseRaw, Epochs, Evoked, BaseSpectrum)
+    for inst in insts:
+        _validate_type(inst, valid_types, "instance type")
+        if type(inst) != inst_type:
+            raise ValueError("All insts must be the same type")
+
+    # check that input is a list
+    if not isinstance(insts, list):
+        raise ValueError(f"insts must be a *list* of mne objects, got {type(insts)}")
+
+    # check that all channels have the same name and same number
+    ch_names = insts[0].info.ch_names
+    diff_chns = []
+    for inst in insts[1:]:
+        if inst.info.ch_names != ch_names:
```

Comment:
```suggestion
        if inst.ch_names != ch_names:
```

Reply:
check the set(ch_names) equality first, if the sets are the same but the orders are different, give an appropriate error msg. Error msgs should reference how to fix it (e.g., `equalize_channels` func)